// @webstudio-is/sdk-components-react/lib/time.js
import { jsx as m } from "react/jsx-runtime";
import { forwardRef as a } from "react";
const N = [
  "af",
  "am",
  "ar",
  "az",
  "be",
  "bg",
  "bn",
  "bs",
  "ca",
  "cs",
  "cy",
  "da",
  "de",
  "el",
  "en",
  "es",
  "et",
  "eu",
  "fa",
  "fi",
  "fr",
  "ga",
  "gl",
  "gu",
  "he",
  "hi",
  "hr",
  "hu",
  "hy",
  "id",
  "is",
  "it",
  "ja",
  "ka",
  "kk",
  "km",
  "kn",
  "ko",
  "ky",
  "lb",
  "lt",
  "lv",
  "mk",
  "ml",
  "mn",
  "mr",
  "ms",
  "mt",
  "nb",
  "nl",
  "nn",
  "pl",
  "pt",
  "ro",
  "ru",
  "si",
  "sk",
  "sl",
  "sq",
  "sr",
  "sv",
  "sw",
  "ta",
  "te",
  "th",
  "tr",
  "uk",
  "ur",
  "uz",
  "vi",
  "zh"
], E = [
  "AF",
  "AL",
  "DZ",
  "AS",
  "AD",
  "AO",
  "AI",
  "AQ",
  "AG",
  "AR",
  "AM",
  "AW",
  "AU",
  "AT",
  "AZ",
  "BS",
  "BH",
  "BD",
  "BB",
  "BY",
  "BE",
  "BZ",
  "BJ",
  "BM",
  "BT",
  "BO",
  "BA",
  "BW",
  "BR",
  "BN",
  "BG",
  "BF",
  "BI",
  "CV",
  "KH",
  "CM",
  "CA",
  "KY",
  "CF",
  "TD",
  "CL",
  "CN",
  "CO",
  "KM",
  "CG",
  "CD",
  "CR",
  "HR",
  "CU",
  "CY",
  "CZ",
  "DK",
  "DJ",
  "DM",
  "DO",
  "EC",
  "EG",
  "SV",
  "GQ",
  "ER",
  "EE",
  "SZ",
  "ET",
  "FJ",
  "FI",
  "FR",
  "GA",
  "GM",
  "GE",
  "DE",
  "GH",
  "GR",
  "GD",
  "GT",
  "GN",
  "GW",
  "GY",
  "HT",
  "HN",
  "HU",
  "IS",
  "IN",
  "ID",
  "IR",
  "IQ",
  "IE",
  "IL",
  "IT",
  "JM",
  "JP",
  "JO",
  "KZ",
  "KE",
  "KI",
  "KP",
  "KR",
  "KW",
  "KG",
  "LA",
  "LV",
  "LB",
  "LS",
  "LR",
  "LY",
  "LI",
  "LT",
  "LU",
  "MG",
  "MW",
  "MY",
  "MV",
  "ML",
  "MT",
  "MH",
  "MR",
  "MU",
  "MX",
  "FM",
  "MD",
  "MC",
  "MN",
  "ME",
  "MA",
  "MZ",
  "MM",
  "NA",
  "NR",
  "NP",
  "NL",
  "NZ",
  "NI",
  "NE",
  "NG",
  "NO",
  "OM",
  "PK",
  "PW",
  "PA",
  "PG",
  "PY",
  "PE",
  "PH",
  "PL",
  "PT",
  "QA",
  "RO",
  "RU",
  "RW",
  "KN",
  "LC",
  "VC",
  "WS",
  "SM",
  "ST",
  "SA",
  "SN",
  "RS",
  "SC",
  "SL",
  "SG",
  "SK",
  "SI",
  "SB",
  "SO",
  "ZA",
  "SS",
  "ES",
  "LK",
  "SD",
  "SR",
  "SE",
  "CH",
  "SY",
  "TW",
  "TJ",
  "TZ",
  "TH",
  "TL",
  "TG",
  "TO",
  "TT",
  "TN",
  "TR",
  "TM",
  "TV",
  "UG",
  "UA",
  "AE",
  "GB",
  "US",
  "UY",
  "UZ",
  "VU",
  "VA",
  "VE",
  "VN",
  "YE",
  "ZM",
  "ZW"
], G = "dateTime attribute is not set", L = "", M = "en", u = "GB", f = "medium", Y = "none", g = (t) => N.includes(t) ? t : M, I = (t) => E.includes(t) ? t : u, B = (t) => {
  if (["full", "long", "medium", "short"].includes(t))
    return t;
}, R = (t) => {
  if (["full", "long", "medium", "short"].includes(t))
    return t;
}, C = (t) => {
  if (t === "")
    return;
  let n = new Date(t);
  if (Number.isNaN(n.getTime()) === !1)
    return n;
  if (/^\d+$/.test(t)) {
    let e = Number(t);
    t.length === 10 && (e *= 1e3), n = new Date(e);
  }
  if (Number.isNaN(n.getTime()) === !1)
    return n;
}, U = (t, n) => {
  const e = (s, r = 2) => String(s).padStart(r, "0"), o = {
    YYYY: t.getFullYear(),
    YY: String(t.getFullYear()).slice(-2),
    MM: e(t.getMonth() + 1),
    M: t.getMonth() + 1,
    DD: e(t.getDate()),
    D: t.getDate(),
    HH: e(t.getHours()),
    H: t.getHours(),
    mm: e(t.getMinutes()),
    m: t.getMinutes(),
    ss: e(t.getSeconds()),
    s: t.getSeconds()
  };
  return n.replace(
    /\b(YYYY|YY|MM|M|DD|D|HH|H|mm|m|ss|s)\b/g,
    (s) => String(o[s])
  );
}, h = a(
  ({
    language: t = M,
    country: n = u,
    dateStyle: e = f,
    timeStyle: o = Y,
    format: s,
    // native html attribute in react style
    dateTime: r = G,
    ...c
  }, S) => {
    const A = `${g(t)}-${I(
      n
    )}`, D = {
      dateStyle: B(e),
      timeStyle: R(o)
    }, i = r === null ? L : r.toString(), l = C(i);
    let T = i;
    if (l)
      if (s)
        try {
          T = U(l, s);
        } catch {
        }
      else
        try {
          T = new Intl.DateTimeFormat(A, D).format(l);
        } catch {
        }
    return /* @__PURE__ */ m("time", { ref: S, dateTime: i, ...c, children: T });
  }
);
export {
  h as Time
};
